tests/tests_pytorch/loops/test_fetchers.py

```diff
@@ -192,12 +195,14 @@ def training_step(self, dataloader_iter, batch_idx):
                 opt.step()
 
         def on_train_epoch_end(self):
-            assert self.trainer.fit_loop.epoch_loop.batch_progress.current.ready == 33
+            # since the dataset is sized, the loop stops at the limit even though the training_step controls the
+            # consumption of batches
```

Comment:
Does that mean we no longer enter the *_step() method where the StopIteration would normally occur? (for the sized iterator case)

Reply:
That's right. This also means https://docs.google.com/document/d/1DQxvUdkxdh3jmWorA5g12LeOOU0tcJ7d0uqN5LyvISQ/edit?disco=AAAA4XIqT6M